java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/JniSinkWriterResponseObserver.java

```diff
@@ -25,22 +26,24 @@ public class JniSinkWriterResponseObserver
     private static final Logger LOG = LoggerFactory.getLogger(JniSinkWriterResponseObserver.class);
     private long responseTxPtr;
 
-    private boolean success;
+    private boolean success = true;
 
     public JniSinkWriterResponseObserver(long responseTxPtr) {
         this.responseTxPtr = responseTxPtr;
     }
 
     @Override
     public void onNext(ConnectorServiceProto.SinkWriterStreamResponse response) {
-        this.success =
-                Binding.sendSinkWriterResponseToChannel(this.responseTxPtr, response.toByteArray());
+        if (!Binding.sendSinkWriterResponseToChannel(this.responseTxPtr, response.toByteArray())) {
+            throw Status.INTERNAL.withDescription("unable to send response").asRuntimeException();
+        }
     }
 
     @Override
     public void onError(Throwable throwable) {
         this.success =
                 Binding.sendSinkWriterErrorToChannel(this.responseTxPtr, throwable.getMessage());
```

Comment:
nits: we don't need to assign `this.success` here.

Also, shall we throw exception here if the channel is close to make it consistent with `onNext`?

Reply:
fixed.

In `onError` setting the success flag shall be enough. The handler will stop when it detects that the flag is set false. I have added a warning log to print the unsent error message.